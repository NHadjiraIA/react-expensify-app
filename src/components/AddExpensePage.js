import React , {useState} from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm'
import { addExpense} from '../actions/exprenses'
 

export class AddExpensePage extends React.Component {
        onSubmit = (expense) =>{
            //props.dispatch(addExpense(expense));
            this.props.addExpense(expense)
            //to redurect to other page
            this.props.history.push('/')
        }
        render() {
            return (
                <div>
                    <h1> Add Expense</h1>
                    <ExpenseForm
                    onSubmit={this.onSubmit}
                    /> 
                </div>
                
         )}
}
 
const mapDispatchToProps =(dispatch) =>{
    return{
        addExpense : (expense) => dispatch(addExpense(expense))
    }
}
   export default connect(undefined, mapDispatchToProps)(AddExpensePage);