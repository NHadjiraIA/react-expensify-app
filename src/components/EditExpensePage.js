import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, removeExpense} from '../actions/exprenses'

export class  EditExpensePage extends React.Component {
  onSubmit = (expense) =>{
    //dispatch the action to edit the expense
   // props.dispatch(editExpense(props.expense.id, expense));
    this.props.editExpense(this.props.expense.id,expense);
    //redirect to the dashbord
    this.props.history.push('/');
   }
   onRemove = () => {
    //props.dispatch(removeExpense({id: props.expense.id}))
    this.props.removeExpense({id: this.props.expense.id})
    this.props.history.push('/')
}
render (){
  return (
    <div>
         <ExpenseForm 
         // to pass the props from dashbord to edit form
         expense={this.props.expense}
         onSubmit={this.onSubmit}
         />
          <button onClick={this.onRemove}> Remove </button>
      </div>
  )
}
}
const mapDispatchToProps =(dispatch,props) =>{
  return({
    editExpense : (id,expense) => dispatch(editExpense(id,expense)),
    removeExpense : (data) => dispatch(removeExpense(data))
  });
};
 
const mapStateToProps = (state, props) =>{
   return{
      expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    }
}
export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);