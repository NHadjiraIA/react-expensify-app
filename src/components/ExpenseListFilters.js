import React from "react";
import { connect } from "react-redux";
import { setTextFilter } from '../actions/filters'
import { sortByAmount, sortByDate, setStartDate, setEndDate } from '../actions/filters'
import { DateRangePicker } from 'react-dates'
// we need to swich the fonction component to  a class based component because we have to use state
export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null
    }
    onDatesChange = ({startDate, endDate}) =>{
     this.props.setStartDate(startDate);
     this.props.setEndDate(endDate);
    }
    onFocusChange =(calendarFocused) =>{
        this.setState (()=>({calendarFocused}))
    }
    onTextChange = (e) =>{
        this.props.setTextFilter(e.target.value)
    }
    onSortChange = (e) =>{
        {
            if (e.target.value === 'date'){
                this.props.sortByDate()
            } else if (e.target.value === 'amount'){
                this.props.sortByAmount()
            }
       }
    }
    render () {
        return(
        <div>
        <input 
        type='text' 
        value ={this.props.filters.text}
         onChange={ this.onTextChange}
         />
          <select 
          value ={this.props.filters.sortBy} 
          onChange={ this.onSortChange} >
           <option value='date'> Date </option>
           <option value='amount'> Amount </option>
         </select>

         <DateRangePicker 
           startDate={this.props.filters.startDate}
           endDate={this.props.filters.endDate}
           onDatesChange={this.onDatesChange}
           focusedInput ={this.state.calendarFocused}
           onFocusChange ={this.onFocusChange}
           showClearDates={true}
           numberOfMonths={1}
           isOutsideRange={() => false}
         />
    </div>
    )}
}

const  mapDispatchToProps = (dispatch) =>({
    setTextFilter : (text)=> dispatch(setTextFilter(text)),
    sortByDate : () => dispatch(sortByDate()),
    sortByAmount : () =>dispatch(sortByAmount()),
    setStartDate : (startDate) => dispatch(setStartDate(startDate)),
    setEndDate : (endDate) =>dispatch(setEndDate(endDate))
});

const mapStateToProps = (state) => {
    return {
        filters : state.filters
    };
};
export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);