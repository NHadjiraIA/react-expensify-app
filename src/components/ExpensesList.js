import React from "react";
import { connect } from "react-redux";
import ExpenseListItem from "./ExpenseListItem";
import selectorExpenses from '../selectors/expenses';

// we can do like this in line 15 to 19
/*{props.expenses.map((expense) => 
    <ExpenseListItem 
      key = {expense}
      description = {expense.description}
      amount = {expense.amount}
    />) } */
// this is the component that we want to connect to redux
export const ExpensesList = (props) => (
    <div>
        <h1> Expenses List</h1>
        <div>
            {
                props.expenses.length === 0 ? (
                   <p> No exepenses </p>
                ) : (
                    props.expenses.map((expense) => {
                        return <ExpenseListItem key = {expense.id}{...expense} />     
                     })
                 )
            }
        </div>
       
    </div>
);

// this is the  store of our app
const mapsStateToProps =  (state) => {
    return {
       expenses: selectorExpenses(state.expenses, state.filters)
    }
   }
   // here we connect the store to the component means that we connect state to props
export default  connect(mapsStateToProps)(ExpensesList);
 