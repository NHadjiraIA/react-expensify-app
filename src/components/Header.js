import React from 'react';
import { NavLink} from 'react-router-dom';

const Header =() =>(
    <div>
        <h1>Expensify</h1>
        <NavLink to='/' activeClassName='is-active' exact={true}> Go home </NavLink>
        <NavLink to='/create' activeClassName='is-active'> Go Create </NavLink>
        <NavLink to='/edit/:id' activeClassName='is-active'> Go Edit </NavLink>
        <NavLink to='/help' activeClassName='is-active'> Go help </NavLink>
    </div>
)

export default Header;