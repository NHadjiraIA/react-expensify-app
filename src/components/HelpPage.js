import React from 'react';

  const HelpPage = () =>(
    <div>
        This the Help Page
    </div>
)

export default HelpPage;