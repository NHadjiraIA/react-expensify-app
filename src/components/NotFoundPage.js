import React from 'react';
import { Link } from 'react-router-dom';

   // when we use <a> to go home page the page refrech but if we use <Link> we switch to home page without refrech 
   const NotFoundPage = () =>(
    <div>
       404 - <Link to='/'> Go home </Link>
    </div>
)

export default NotFoundPage;