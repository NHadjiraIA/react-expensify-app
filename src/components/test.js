console.log("T")
setTimeout(() =>{
    console.log("HHHH")
},0)

let rainF = ["dd","eee","rr"]
rainF.splice(0,2)
console.log(rainF)
const person ={
    age: "dave",
    age:50,
    paye:"canada"
}
// var result = map((x) => x.toUpperCase())
// console.log("this is result", result)
let scors =[]
scors.push(1,2)
scors.pop()
scors.push(3,4)
let scor = scors.reduce( (a,b) => a+b)
console.log("scor", scor)

function sum (num1, num2= 2, num3 =3){
    return num1+ num2+ num3
}
const value = [3.5]
let total = value + sum()
console.log ("total" , total)
const arr1 = [2,4,6]
const arr2 = [3,5,7]
console.log([...arr1, ...arr2])
