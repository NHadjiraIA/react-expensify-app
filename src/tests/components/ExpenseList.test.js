import React from "react";
import { shallow } from "enzyme";
import { ExpensesList} from '../../components/ExpensesList'
import  expenses from '../fixtures/expenses';

test('should render  ExpensList with expenses', () =>{
  const wrapper = shallow(<ExpensesList  expenses={expenses}/>)
  expect(wrapper).toMatchSnapshot();
 
});


test('should render  ExpensList with empty message', () =>{
    const wrapper = shallow(<ExpensesList  expenses={[]}/>)
    expect(wrapper).toMatchSnapshot();
   
  });

