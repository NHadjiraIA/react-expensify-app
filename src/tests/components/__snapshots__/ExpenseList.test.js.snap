// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render  ExpensList with empty message 1`] = `
<div>
  <h1>
     Expenses List
  </h1>
  <div>
    <p>
       No exepenses 
    </p>
  </div>
</div>
`;

exports[`should render  ExpensList with expenses 1`] = `
<div>
  <h1>
     Expenses List
  </h1>
  <div>
    <ExpenseListItem
      amount={123}
      createdAt={0}
      description="Gaz"
      id="1"
      note="this is  1st note"
    />
    <ExpenseListItem
      amount={300}
      createdAt={-345600000}
      description="Rent"
      id="2"
      note="this is  2nd note"
    />
    <ExpenseListItem
      amount={500}
      createdAt={345600000}
      description="Credit card"
      id="3"
      note="this is  3rd note"
    />
  </div>
</div>
`;
